index.js

```diff
@@ -164,18 +149,45 @@ function buildInfo(source) {
 }
 
 /*
- * Assigns host, basePath and schemes for Swagger 2.0 result document from
+ * Get host, basePath and schemes for Swagger 2.0 result document from
  * Swagger 1.2 basePath.
  * @param basePath {string} - the base path from Swagger 1.2
- * @param result {object} - Swagger 2.0 document
 */
-function assignPathComponents(basePath, result) {
+function buildPathComponents(basePath) {
+  if (!basePath) { return {}; }
+
   var url = urlParse(basePath);
-  result.host = url.host;
-  result.basePath = url.path;
-  if (url.protocol) {
-    result.schemes = [url.protocol.substr(0, url.protocol.length - 1)];
-  }
+  return {
+    host: url.host,
+    basePath: url.path || '/',
+    //url.protocol include traling colon
+    schemes: url.protocol && [url.protocol.slice(0, -1)]
+  };
+}
+
+function buildTypeProperties(oldType) {
+  if (!oldType) { return {}; }
+
+  //TODO: handle list[<TYPE>] types.
+  //if () {
+  //  return {type: 'array', 'items': {type: }};
+  //}
+
+  return {
+    integer: {type: 'integer'},
+    number:  {type: 'number'},
+    string:  {type: 'string'},
+    boolean: {type: 'boolean'},
+    array:   {type: 'array'},
+    file:    {type: 'file'},
+    int:     {type: 'integer', format: 'int32'},
+    long:    {type: 'integer', format: 'int64'},
+    float:   {type: 'number',  format: 'float'},
+    double:  {type: 'double',  format: 'double'},
+    date:    {type: 'string',  format: 'date-time'},
+    list:    {type: 'array'},
+    void:    {}
+  }[oldType.toLowerCase()] || {$ref: oldType};
```

Comment:
Yes, you can.
But I will rewrite this code, and create separate variable for object.
Because it hard to read and there really big distance between `return` and `;`

P.S. It actually common trick from Python language :)